api/src/main/java/org/apache/iceberg/Files.java

```diff
@@ -95,26 +95,40 @@ public String toString() {
   }
 
   public static InputFile localInput(File file) {
-    return new LocalInputFile(file);
+    return new LocalInputFile(file, -1L);
+  }
+
+  public static InputFile localInput(File file, long length) {
+    return new LocalInputFile(file, length);
   }
 
   public static InputFile localInput(String file) {
+    return localInput(file, -1L);
```

Comment:
I don't think it is a good idea to use -1. Can you refactor so that it isn't required?

Reply:
Sure, will do.